src/main/java/uk/gov/companieshouse/util/EricHeaderParser.java

```diff
@@ -0,0 +1,101 @@
+package uk.gov.companieshouse.util;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.model.CreatedBy;
+
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+
+@Component
+public class EricHeaderParser {
+
+    private static final String DELIMITER = ";";
+    private static final String EMAIL_IDENTIFIER = "@";
+    private static final String ERIC_FORENAME_KEY = "forename=";
+    private static final String ERIC_FORENAME_KEY_UTF8 = "forename*=";
+    private static final String ERIC_SURNAME_KEY = "surname=";
+    private static final String ERIC_SURNAME_KEY_UTF8 = "surname*=";
+
+
+    public CreatedBy retrieveUser(String userId, String authorisedUser) {
+
+        CreatedBy createdBy = new CreatedBy();
+        createdBy.setId(userId);
+        createdBy.setForename(getForename(authorisedUser));
```

Comment:
Didn't we say that we do not want to process forename, surname and email?

Reply:
i'm going to remove this now